Scripts/RemediationScripts/Remediate-EnableAADForKubernetesService.ps1

```diff
@@ -0,0 +1,564 @@
+        $SubscriptionId,
+
+        [Switch]
+        [Parameter(ParameterSetName = "DryRun", HelpMessage="Specifies validation of prerequisites for the command")]
+        [Parameter(ParameterSetName = "WetRun", HelpMessage="Specifies validation of prerequisites for the command")]
+        $PerformPreReqCheck,
+
+        [Switch]
+        [Parameter(ParameterSetName = "DryRun", Mandatory = $true, HelpMessage="Specifies a dry run of the actual remediation")]
+        $DryRun,
+
+        [String]
+        [Parameter(ParameterSetName = "WetRun", HelpMessage="Specifies the path to the file to be used as input for the remediation")]
+        $FilePath
+    )
+
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "`n[Step 1 of 5] Preparing to enable AAD for Kubernetes Services in Subscription: $($SubscriptionId)"
+
+    if ($PerformPreReqCheck)
+    {
+        try
+        {
+            Write-Host "Setting up prerequisites..."
+            Setup-Prerequisites
+        }
+        catch
+        {
+            Write-Host "Error occurred while setting up prerequisites. Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+            break
+        }
+    }
+
+    # Get current Azure account context.
+    $context = Get-AzContext
+
+    if ([String]::IsNullOrWhiteSpace($context))
+    {
+        Write-Host "No active Azure login session found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+        break
+    }
+
+    # Setting up context for the current Subscription.
+    $context = Set-AzContext -SubscriptionId $SubscriptionId -ErrorAction Stop
+    
+    Write-Host $([Constants]::SingleDashLine)
+    Write-Host "Subscription Name: $($context.Subscription.Name)"
+    Write-Host "Subscription ID: $($context.Subscription.SubscriptionId)"
+    Write-Host "Account Name: $($context.Account.Id)"
+    Write-Host "Account Type: $($context.Account.Type)"
+    Write-Host $([Constants]::SingleDashLine)
+
+    Write-Host "*** To enable AAD for Kubernetes Services in a Subscription, Contributor or higher privileges on the Kubernetes Services are required. ***" -ForegroundColor $([Constants]::MessageType.Info)
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "`n[Step 2 of 5] Preparing to fetch all Kubernetes Services..."
+
+    $kubernetesServiceResourceType = "Microsoft.ContainerService/managedClusters"
+    $kubernetesServiceResources = @()
+
+    # No file path provided as input to the script. Fetch all Kubernetes Services in the Subscription.
+    if ([String]::IsNullOrWhiteSpace($FilePath))
+    {
+        Write-Host "Fetching all Kubernetes Services in Subscription: $($context.Subscription.SubscriptionId)" -ForegroundColor $([Constants]::MessageType.Info)
+
+        # Get all Kubernetes Services in a Subscription
+        $kubernetesServiceResources = Get-AzResource -ResourceType $kubernetesServiceResourceType -ErrorAction Stop
```

Comment:
Instead of fetching generic resource type, if use Get-AzAksCluster to fetch all AKS cluster in subscription that might improve the performance.

Reply:
Makes sense. 
Done.